dev/automate_training.py

```diff
@@ -29,48 +34,80 @@
 def get_parser():
     parser = argparse.ArgumentParser()
     parser.add_argument("-c", "--config", required=True, help="Base config file path.")
-    parser.add_argument("--all-combin", dest='all_combin', action='store_true', help="To run all combinations of config")
+    parser.add_argument("-n", "--n-iterations", dest="n_iterations",
+                        type=int, help="Number of times to run each config .")
+    parser.add_argument("--all-combin", dest='all_combin', action='store_true',
+                        help="To run all combinations of config")
+    parser.add_argument("--run-test", dest='run_test', action='store_true',
+                        help="Evaluate the trained model on the testing sub-set.")
+    parser.add_argument("--fixed-split", dest='fixed_split', action='store_true',
+                        help="Keep a constant dataset split for all configs and iterations")
     parser.set_defaults(all_combin=False)
 
     return parser
 
 
-def worker(config):
+def train_worker(config):
     current = mp.current_process()
-    #ID of process used to assign a GPU
+    # ID of process used to assign a GPU
     ID = int(current.name[-1]) - 1
 
-    #Use GPU i from the array specified in the config file
+    # Use GPU i from the array specified in the config file
     config["gpu"] = config["gpu"][ID]
 
-    #Call ivado cmd_train
+    # Call ivado cmd_train
     try:
-        #Save best validation score
-        best_training_dice, best_training_loss, best_validation_dice, best_validation_loss = ivado.cmd_train(config)
+        # Save best validation score
+        best_training_dice, best_training_loss, best_validation_dice, best_validation_loss = ivado.cmd_train(
+            config)
+
     except:
         logging.exception('Got exception on main handler')
         print("Unexpected error:", sys.exc_info()[0])
         raise
 
     # Save config file in log directory
-    config_copy = open(config["log_directory"] + "/config.json","w")
+    config_copy = open(config["log_directory"] + "/config.json", "w")
     json.dump(config, config_copy, indent=4)
 
     return config["log_directory"], best_training_dice, best_training_loss, best_validation_dice, best_validation_loss
 
 
+def test_worker(config):
+    current = mp.current_process()
+    # ID of process used to assign a GPU
+    ID = int(current.name[-1]) - 1
+
+    # Use GPU i from the array specified in the config file
+    config["gpu"] = config["gpu"][ID]
+
+    # Call ivado cmd_eval
+    try:
+        # Save best test score
+
+        config["command"] = "eval"
+        test_dict, eval_df = ivado.cmd_eval(config)
```

Comment:
What about using the 3D dice from `eval_df`?

Reply:
Sorry if it's stupid question, but wouldn't it give the same result?

Reply:
Not necessary, but probably close indeed --> computed by batch vs computed by subjects.